### ISSUE TITLE
cache iterator subclass barfs on iterators that worked with earlier code

### ISSUE THREAD

**Issue by SysPete** [racke/Template-Flute#87]
Commit 8ee7a535c2036d2ce1d2448d1dfd8d6a84d698b2 causes breakage in Interchange6-Demo-Shop product-page view.
Here is the iterator generated using Interchange6::Schema::Result::Product->attribute_iterator:
```
[
  {
    attribute_values => [
      {
        priority => 80,
        selected => 0,
        title => "1/4\"",
        value => "1/4 inch"
      },
      {
        priority => 70,
        selected => 0,
        title => "1/2\"",
        value => "1/2 inch"
      },
      {
        priority => 60,
        selected => 0,
        title => "1\"",
        value => "1 inch"
      },
      {
        priority => 50,
        selected => 0,
        title => "2\"",
        value => "2 inches"
      },
      {
        priority => 40,
        selected => 0,
        title => "3\"",
        value => "3 inches"
      }
    ],
    name => "diameter",
    priority => 1,
    title => "Diameter"
  }
]
```
And an extract from Dancer log:
```
19/Mar/2015 17:51:27 [16503] error @0.298907> request to GET /mechanics-pliers crashed: isa check for "iterator" failed:  is not blessed at (eval 786) line 137.
        eval {...} called at (eval 786) line 135
        Template::Flute::Iterator::Cache::new("Template::Flute::Iterator::Cache", "iterator", "") called at /home/apm/git/Template-Flute/lib/Template/Flute.pm line 577
        Template::Flute::_sub_process(Template::Flute=HASH(0xbfe1698), XML::Twig=HASH(0xc0bd1c0), XML::Twig::Elt=HASH(0xc11f5a0), HASH(0xbdd8ff0), Template::Flute::Specification=HASH(0xa133d78), Template::Flute::HTML=HASH(0xc0b71a8), 0, 0) called at /home/apm/git/Template-Flute/lib/Template/Flute.pm line 481
        Template::Flute::process(Template::Flute=HASH(0xbfe1698)) called at /home/apm/perl5/perlbrew/perls/perl-5.14.4/lib/site_perl/5.14.4/Dancer/Template/TemplateFlute.pm line 488
        Dancer::Template::TemplateFlute::render(Dancer::Template::TemplateFlute=HASH(0xbdd5f70), "/home/apm/git/Interchange6-Demo-Shop/DanceShop/views/product-"..., HASH(0xbdd8ff0)) called at /home/apm/perl5/perlbrew/perls/perl-5.14.4/lib/site_perl/5.14.4/Dancer/Template/Abstract.pm line 82
```

**Comment by SysPete** [comment]
In case you are wondering which product it is from the base demo catalogue:
http://localhost:5000/mechanics-pliers

**Comment by racke** [comment]
Fixed by 6ec72ad.
